### PR TITLE
Fix sidecar provisioner when deploying from charmhub too quickly.

### PR DESCRIPTION
**Pull request by hpidcock**
Delays the provisioning of sidcar app until charm fully downloaded.

Also cleans up some error handling.

## QA steps

```
juju deploy indico
juju debug-log
```

check for no `controller-0: 14:37:28 ERROR juju.charmhub charmhub empty id not valid`

## Documentation changes

N/A

## Bug reference

https://bugs.launchpad.net/juju/+bug/1991048

### CODE REVIEW

File: rpc/params/apierror.go [hunk 1]
```diff
@@ -214,6 +214,45 @@ const (
 	CodeNotValid                  = "not valid"
 )
 
+// TranslateWellKnownError translates well known wire error codes into a github.com/juju/errors error
+// that matches the error code.
+func TranslateWellKnownError(err error) error {
+	code := ErrCode(err)
+	switch code {
+	// case CodeDeadlineExceeded:
+	// 	return errors.NewTimeout(err, "")
```
Comment (wallyworld) [inline rpc/params/apierror.go:223]: ?

File: worker/caasapplicationprovisioner/application.go [hunk 2]
```diff
@@ -697,7 +701,10 @@ func (a *appWorker) alive(app caas.Application) error {
 	a.logger.Debugf("ensuring application %q exists", a.name)
 
 	provisionInfo, err := a.facade.ProvisioningInfo(a.name)
-	if err != nil {
+	if errors.Is(err, errors.NotProvisioned) {
+		a.logger.Debugf("cannot ensure application %s yet: %s", a.name, err.Error())
```
Comment (wallyworld) [inline worker/caasapplicationprovisioner/application.go:705]: maybe better as "application is not provisioned yet"

**Review by wallyworld — APPROVED** [review]

File: rpc/params/apierror.go (hunk 1 above)
Comment (hpidcock) [reply rpc/params/apierror.go:223]: errors package doesn't have a deadline exceeded, not sure timeout is the best error, will add it as a todo

**Pull request merged**